engine/runtime-instrument-common/src/main/java/org/enso/interpreter/service/ExecutionService.java

```diff
@@ -552,12 +597,11 @@ public boolean isExitException(AbstractTruffleException ex) {
   /**
    * Returns a human-readable message for a panic exception.
    *
-   * @param panic the panic to display.
-   * @return a human-readable version of its contents.
+   * @param panic the panic to display
+   * @return a computation of a human-readable version of an exception
    */
-  public String getExceptionMessage(AbstractTruffleException panic) {
-    var future = submitExecution(() -> computeExceptionMessage(panic));
-    return resultOf(future);
+  public CompletionStage<String> getExceptionMessage(AbstractTruffleException panic) {
```

Comment:
This one would very likely work better as:
```java
CompletionStage<String> getExceptionMessage(CompletionStage<AbstractTruffleException> panic)
```
or something a like that being _part of the reactive flow_. It makes little sense to `ex = toCompletableFuture().get()` and then enter again the _guest code_ thread by `getExceptionMessage(ex)`.

The same comment applies to `typeOfValue` and other methods that inspect _result of guest code_ execution.